main.js

```diff
@@ -3,29 +3,37 @@
 
 // Gets branding for an article.
 // Branding can either be an article brand or a columnist
-module.exports = function(metadata) {
-	if (!metadata) {
-		return;
-	}
-	let brandingToUse = {};
-	let columnists = [];
+// Metadat is based on v3 ES
 
-	if (metadata.tags) {
-		columnists = metadata.tags.map(function(tag) {
-			const attrs = tag.term.attributes;
-			return (attrs && attrs.length && attrs[0].key === 'isColumnist' && attrs[0].value === true) ? tag : null;
-		})
-		.filter(function(e) { return e; }); // Clean out falsy vals from the array
+module.exports = function (metadata) {
+	// check for whether a brand tag exists
+	let matchedTag = metadata.find(tag => tag.taxonomy === 'brand');
+	// but Columnist trumps brand
+	if (
+		metadata.find(tag => tag.taxonomy === 'authors') &&
+		metadata.find(tag =>
+			tag.taxonomy === 'genre' &&
+			tag.prefLabel === 'Comment'
+		) &&
+		metadata.find(tag =>
+			tag.taxonomy === 'sections' &&
+			tag.prefLabel === 'Columnists')
+	) {
+		matchedTag = metadata.find(tag => tag.taxonomy === 'authors');
 	}
```

Comment:
The condition and block are hard to read in this format. Maybe assign the result of each find to suitably verbose variable (which would also save repeating the find in the block).